asabiris/orchestration/sendemail.py

```diff
@@ -1,228 +1,83 @@
-				error_details = (
-					"Timestamp: {}\n"
-					"Recipients: {}\n"
-					"Error Message: {}\n"
-				).format(current_timestamp, email_addresses, str(e))
-
-				# User-friendly error message
-				error_message = (
-					"Hello!<br><br>"
-					"We encountered an issue while processing your request. "
-					"Please review your input and try again.<br><br>"
-					"Thank you!<br>"
-					"<br>Error Details:<br><pre style='font-family: monospace;'>{}</pre>".format(
-						error_details
-					)
-				)
-
-				# Add LogMan signature with HTML line breaks
-				error_message += "<br>Best regards,<br>LogMan.io"
-
-				subject = "Error Processing Request"
-
-				return error_message, subject
-
-	def get_file_name(self, attachment):
-			"""
-			This method returns a file-name if provided in the attachment-dict.
-			If not then the name of the file is current date with appropriate
-			extensions.
-			"""
-			if attachment.get('filename') is None:
-				now = datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
-				return "att-" + now + "." + attachment.get('format')
-			else:
-				return attachment.get('filename')
+class SendEmailOrchestrator:
+
+    def __init__(self, app):
+        self.services = {
+            name: app.get_service(name) for name in [
+                "JinjaService", "HtmlToPdfService", "MarkdownToHTMLService", "SmtpService"
+            ]
+        }
+
+    async def send_email(self, email_to: List[str], body_template: str, **kwargs):
+        body_html, email_subject_body = await self._render_template(body_template, kwargs.get('body_params', {}), email_to)
+        atts = self._process_attachments(kwargs.get('attachments', []), email_to) if not asab.Config.get("jinja", "failsafe") else []
+        await self.services['SmtpService'].send(
+            email_from=kwargs.get('email_from'),
+            email_to=email_to,
+            email_cc=kwargs.get('email_cc', []),
+            email_bcc=kwargs.get('email_bcc', []),
+            email_subject=kwargs.get('email_subject', email_subject_body),
+            body=body_html,
+            attachments=atts
+        )
+
+    async def _render_template(self, template: str, params: Dict, email_to: List[str]) -> Tuple[str, str]:
+        try:
+            if not template.startswith("/Templates/Email/"):
+                raise PathError(use_case='Email', invalid_path=template)
+            jinja_output = await self.services['JinjaService'].format(template, params)
+            ext = os.path.splitext(template)[1]
+            if ext not in ['.html', '.md']:
+                raise FormatError(format=ext)
+            return {
+                '.html': utils.find_subject_in_html,
+                '.md': lambda x: (self.services['MarkdownToHTMLService'].format(utils.find_subject_in_md(x)[0]), utils.find_subject_in_md(x)[1])
+            }[ext](jinja_output)
+        except (jinja2.exceptions.TemplateError, Exception) as e:
+            if asab.Config.get("jinja", "failsafe"):
```

Comment:
Failsafe should not be configurable, it should be always enabled.
But thru exception.